File: Practica3/QytetetJava/src/modeloqytetet/Dado.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package modeloqytetet;

import java.util.Random;

/**
 *
 * @author Lorena
 */
public class Dado {
    private static final Dado INSTANCE = new Dado();
    private Dado() {
    }
    
    public static Dado getInstance() {
        return INSTANCE;
    }
    
    //Método
    int tirar(){
        Random aleatorio = new Random();
        int dado = aleatorio.nextInt(6) +1;
        return dado;
    }
}

File: Practica4/QytetetRuby P4/pkg/QytetetRuby-0.0.1/lib/dado.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require 'singleton'

module ModeloQytetet
  class Dado
    include Singleton

    #Método
    def tirar
      dado = Random.new
      dado = rand(1..6)
#      dado = 4
      dado
    end
  end
end
